src/controllers/source-statuses.controller.ts

```diff
@@ -1,43 +1,84 @@
 import { Request, Response } from 'express';
+import { SourceStatusesService } from '../services/source-statuses.service';
 import { SourceStatus } from '../models/source-status.model';
+import { NotFoundError } from '../errors/not-found.error';
+import { NotUniqueError } from '../errors/not-unique.error';
+import { RemovalRestrictedError } from '../errors/removal-restricted.error';
 
 export class SourceStatusesController {
-  static test = async (req: Request, res: Response) => {
-    res.send('Test response');
-  };
-
-  static async getAll(req: Request, res: Response) {
-    const source_statuses = await SourceStatus.find();
-    res.send(source_statuses);
+  static async getSourceStatuses(req: Request, res: Response): Promise<Response<SourceStatusesService>> {
+    try {
+      const source_statuses = await SourceStatusesService.getAll();
+      return res.json({ data: source_statuses });
+    } catch (e) {
+      return res.status(500).json({ message: e.message });
+    }
   }
 
-  static async getOneById(req: Request, res: Response) {
-    const id: string = req.params.id;
+  static async getSourceStatusById(req: Request, res: Response): Promise<Response<SourceStatusesService>> {
+    try {
+      const source_status = await SourceStatusesService.getOneById(Number(req.params.id));
+      return res.json({ data: source_status });
+    } catch (e) {
+      if (e instanceof NotFoundError) {
+        res.status(404).json({ message: e.message });
+      }
+      return res.status(500).json({ message: e.message });
+    }
+  }
 
+  static async createSourceStatus(req: Request, res: Response): Promise<Response<SourceStatusesService>> {
```

Comment:
Incorrect return types in most methods. Correct ones can be peeked at `sources.controller.ts`.